content/token22-metadata.md

```diff
@@ -0,0 +1,1083 @@
+- Create an NFT with metadata embedded in the mint account itself
+- Create an NFT with the metadata pointer extension
+---
+
+# Summary
+- The `metadata pointer` extension associates a public key directly to a metadata account, whether it's stored internally or via an external metadata account like Metaplex. This setup simplifies the association of tokens with their metadata.
+- The `metadata` mint extension allows embedding of metadata directly into mint accounts through the Token Extension Program, typically utilized alongside a self-referencing metadata pointer. This facilitates embedding comprehensive token information at the minting stage.
+- These extensions enhance the interoperability of tokens across different applications and platforms by standardizing how metadata is associated and accessed.
+- Directly embedding or pointing to metadata can streamline transactions and interactions by reducing the need for additional lookups or external calls.
+
+# Overview
+
+The Token Extension Program streamlines metadata on Solana. Without the Token Extension Program, developers store metadata in metadata accounts using a metadata on-chain program; mainly `Metaplex`. However, this has some drawbacks. For example the mint account to which the metadata is "attached" has no awareness of the metadata account. To determine if an account has metadata, we have to PDA the mint and the `Metaplex` program together and query the network to see if a Metadata account exists. Additionally, to create and update this metadata you have to use a secondary program (i.e. `Metaplex`). These processes introduces vender lock and increased complexity. Token Extension Programs's Metadata extensions fix this by introducing two extensions:
+
+- `metadata-pointer` extension: Adds two simple fields in the mint account itself: a publicKey pointer to the account that holds the metadata for the token following the [Token-Metadata Interface](https://github.com/solana-labs/solana-program-library/tree/master/token-metadata/interface), and the authority to update this pointer.
+- `metadata` extension: Adds the fields described in the [Token-Metadata Interface](https://github.com/solana-labs/solana-program-library/tree/master/token-metadata/) which allows us to store the metadata in the mint itself.
+
+Note: The `metadata` extension is usually used in conjunction with the `metadata-pointer` extension which points back to the mint itself.
+
+## Metadata-Pointer extension:
+
+Since multiple metadata programs exist, a mint can have numerous accounts claiming to describe the mint, making it complicated to know which one is the mint's "official" metadata. To resolve this, the `metadata-pointer` extension adds a `publicKey` field to the mint account called `metadataAddress`, which points to the account that holds the metadata for this token. To avoid imitation mints claiming to be a stablecoin, a client can now check whether the mint and the metadata point to each other.
```

Comment:
Might have covered this in my last comment, but just to iterate: "a client can now check whether the mint and the metadata point to each other" if we're already using metadata extension to ensure everything is on the mint, the purpose of check whether the mint and the metadata point to each other seems moot because they are the same account.

Reply:
Mints with `metadata` need the `metadata pointer`

```
    'Program log: A mint with metadata must have the metadata-pointer extension initialized',
    'Program log: Mint or account is initialized to an invalid combination of extensions',
```